k6/docker-entrypoint.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/sh
+set -e
+
+{
+
+# curl_pushgateway_test() tests for a valid PUSHGATEWAY_URL by attempting a
+# POST of a test metric, checking for a valid curl return code, a valid
+# HTTP response code, and lastly a successful query of the test metric value.
+curl_pushgateway_test() {
+  HTTP_CODE=$(echo "test_metric 42" | curl -s --connect-timeout 5 -w "%{http_code}" --data-binary @- "$@" &2>1)
+
+  if  [ $? -ne 0 ]; then
+    echo "PUSHGATEWAY_URL was specified but test connection to ${PUSHGATEWAY_URL} failed. Exiting. "
+    return 1
+  fi
+
+# curl can sometimes return text along with the http code.
+  if [[ ! "${HTTP_CODE}" =~ '^[0-9]+$' ]]; then
+    echo "Pushgateway returned a non-numeric error code (${HTTP_CODE}). "
+    return 1
+  elif [[  ${HTTP_CODE} -lt 200 || ${HTTP_CODE} -gt 299 ]] ; then
+    echo "Pushgateway returned HTTP error (${HTTP_CODE}). "
+    return 1
+  fi
+
+  echo "Verifying POSTED test metric..."
+  QUERY_BASE=$(echo $PUSHGATEWAY_URL | cut -d\/ -f1-3)
+  TEST_VALUE=$(curl -s "${QUERY_BASE}/api/v1/metrics" | jq -r '.data[0].test_metric.metrics[0].value')
+  printf "Queried test mertric value: \"${TEST_VALUE}\" ..."
+  if  [ $TEST_VALUE -ne 42 ]; then
+    echo " FAILED!"
+    echo "\tTest metric POST was successful but test metric verification query failed. Exiting. "
+    return 1
+  fi
+  echo " PASSED!"
+}
```

Comment:
I think I would have just gone with something like `curl --fail --request-target '/-/ready' $PUSHGATEWAY_URL` to test if the Push Gateway was ready, this seems excessive. I'm not going to hold up merging, but I'm also not going to say we aren't going to change it in the future to "fix" the issue of posting dummy metrics.